tests/unit/cli/storage_layout/test_storage_layout_overrides.py

```diff
@@ -435,3 +435,20 @@ def test_module_overlap(make_input_bundle):
             input_bundle=input_bundle,
             storage_layout_override=override,
         )
+
+
+def test_override_with_nonreentrant_pragma(make_input_bundle):
+    code = """
+# pragma nonreentrancy on
+a: public(uint256)
+    """
+
+    if version_check(begin="cancun"):
+        override = {"a": {"type": "uint256", "n_slots": 1, "slot": 0}}
+    else:
+        override = {
+            "a": {"type": "uint256", "n_slots": 1, "slot": 0},
+            "$.nonreentrant_key": {"type": "nonreentrant lock", "n_slots": 1, "slot": 20},
+        }
+
+    compile_code(code, storage_layout_override=override)
```

Comment:
do you want to test round trip here?

Reply:
discussed offline -- it's sanity checked here https://github.com/vyperlang/vyper/blob/14932662d73711524e06a90699b084cf1eea8fff/vyper/semantics/analysis/data_positions.py#L31